network/dropper.go

```diff
@@ -54,9 +54,11 @@ func handleDropping(log zerolog.Logger, wg *sync.WaitGroup, cfg *Config, peers p
 		addresses := peers.Addresses()
 		address := addresses[rand.Int()%len(addresses)]
 		err := peers.Drop(address)
+
 		if err != nil {
 			log.Error().Err(err).Msg("could not drop peer")
 			continue
 		}
+		eventMgr.Disconnected(address)
```

Comment:
It really doesn't make sense to call this here. This is just a handler that causes us to disconnect some peers. It covers a small amount of disconnects. We should have it called somewhere where it works for all of them in a single place.

Reply:
Thanks for the detailed explanation about cascade for peer shutdown.

Reply:
Good point, removed it.